src/luxonis_ml/scheme/v1/config_building_blocks/enums/decoding.py

```diff
@@ -0,0 +1,30 @@
+from enum import Enum
+
+class DecodingFamily(Enum):
+
+    """ members of a family should have similar decoding """
+
+    # classification
+    MULTICLASS_CLASSIFICATION= "multiclass_classification"
+
+    # object detection
+    YOLO = "yolo"
+    SSD_PARSED = "ssd-mobilenet-parsed" # seems like outputs of all ssd-mobilenet networks are of shape (1,3000,classes_n) and (1,3000,4)
```

Comment:
Does current metadata required for decoding Yolo and SSD (called MobileNet) require similar settings?
I am thinking if it would make sense ditching the decoding family all together and only relate on the head settings, where head can have different subtypes.
In case we ditch it, it might make sense to have some unchangeable name attribute in each head?

Reply:
Other than anchors, more or less similar settings, yes. I guess that DecodingFamily doesn't really add much info at this point (e.g. we could easily differentiate between Yolo and SSD decoding by the existing ObjectDetectionSubtype enum). Should I remove it?

What exactly do you mean by the unchangeable name attribute in each head (and how would be best to implement it)?

Reply:
Okay, then I think if it can be inferred based on the subtype, I think we can remove it.

By unchangeable name attribute I meant something along the lines of a constant (https://github.com/pydantic/pydantic/issues/1927).

Basically, each Head metadata would contain information like "DetectionHead" as the main type. Those would be pre-configured inside each head.

Reply:
I added a SSD subtype and removed the DecodingFamily enum. 
With regards adding a static and immutable field to each HeadMetadata subclass, I am wondering what it adds on top of the label_type field (Enum with possible values: classification, object_detection, segmentation, keypoint_detection, or raw)